### PR TITLE
docs: testsuite and emul 3.2 release notes

### PR DESCRIPTION
**Pull request by yperess**
Add release notes for both the emulation and test suite subsystems

### CODE REVIEW

File: doc/releases/release-notes-3.2.rst [hunk 1]
```diff
@@ -1395,6 +1400,15 @@ Libraries / Subsystems
 
 * Testsuite
 
+  * Added Kconfig support to ``unit_testing`` platform.
+  * Migrate tests to use :kconfig:option:`CONFIG_ZTEST_NEW_API`
+  * Add ztest options for shuffling tests/suites via:
+    * :kconfig:option:`CONFIG_ZTEST_SHUFFLE`
```
Comment (fabiobaltieri) [inline doc/releases/release-notes-3.2.rst:1406]: I think that doxygen wants an emptyline before indentation changes, so one here and one after this block.

Comment (yperess) [reply doc/releases/release-notes-3.2.rst:1406]: Done

File: doc/releases/release-notes-3.2.rst [hunk 2]
```diff
@@ -1285,6 +1285,11 @@ Libraries / Subsystems
 
 * Emul
 
+  * Added :c:macro:`EMUL_DT_DEFINE` and :c:macro:`EMUL_DT_INST_DEFINE` to mirror
+    :c:macro:``DEVICE_DT_DEFINE` and :c:macro:`DEVICE_DT_INST_DEFINE` respectively.
```
[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply doc/releases/release-notes-3.2.rst:1289]: yeah the DEVICE_DT_DEFINE one looks wrong

[the remark this answers is not in the document]
Comment (yperess) [reply doc/releases/release-notes-3.2.rst]: Done

**Review by fabiobaltieri — APPROVED** [review]

**Review by asemjonovs — APPROVED** [review]

**Pull request merged**